Realm/Realm.Weaver/RealmWeaver.cs

```diff
@@ -310,15 +313,16 @@ private static void ReplaceGeneratedClassGetter(PropertyDefinition prop, TypeDef
             //// This is equivalent to:
             ////   get => Accessor.Property;
 
-            var start = prop.GetMethod.Body.Instructions.First();
             var il = prop.GetMethod.Body.GetILProcessor();
+            prop.GetMethod.Body.Instructions.Clear();
```

Comment:
I believe the previous implementation of replacing the getter was causing some unintended behavior, it was actually just adding instructions to the getter rather than removing the old instructions---the reason why this was not noticeable was because the old instruction because unreachable because of the `ret`/`return` statement. My ILSpy was actually giving me an error about this prior. One issue this was causing was it kept instructions that have a reference to the backing field, making it error-prone when removing them. This change makes it identical to the functionality in the `ReplaceGeneratedClassSetter`.

Reply:
Good eye! Need to take a look at this